Composer/packages/client/src/pages/design/exportSkillModal/content/Description.tsx

```diff
@@ -50,42 +60,94 @@ const InlineLabelField: React.FC<FieldProps> = (props) => {
+  const { $id, $schema, ...rest } = value;
 
-  const { hidden, properties } = useMemo(
+  const { hidden, properties } = useMemo(() => {
+    if (!schema.properties) return { hidden: [], properties: {} } as any;
+    return Object.entries(schema.properties as JSONSchema7).reduce(
+      ({ hidden, properties }, [key, property]) => {
+        if (property.type === 'object' || (property.type === 'array' && property?.items?.type !== 'string')) {
+          return { hidden: [...hidden, key], properties };
+        }
+
+        const itemSchema = property?.items as JSONSchema7;
+        const serializer =
+          itemSchema?.type === 'string'
+            ? {
+                get: (value) => (Array.isArray(value) ? value.join(',') : value),
+                set: (value) => (typeof value === 'string' ? value.split(/\s*,\s*/) : value),
+              }
+            : null;
+
+        return {
+          hidden,
+          properties: { ...properties, [key]: { field: InlineLabelField, hideError: true, serializer } },
+        };
+      },
+      { hidden: [], properties: {} } as any
+    );
+  }, [schema]);
+
+  const options: IDropdownOption[] = useMemo(
     () =>
-      Object.entries(schema?.properties as JSONSchema7).reduce(
-        ({ hidden, properties }, [key, property]) => {
-          if (property.type === 'object' || (property.type === 'array' && property?.items?.type !== 'string')) {
-            return { hidden: [...hidden, key], properties };
-          }
-
-          const itemSchema = property?.items as JSONSchema7;
-          const serializer =
-            itemSchema?.type === 'string'
-              ? {
-                  get: (value) => (Array.isArray(value) ? value.join(',') : value),
-                  set: (value) => (typeof value === 'string' ? value.split(/\s*,\s*/) : value),
-                }
-              : null;
-
-          return {
-            hidden,
-            properties: { ...properties, [key]: { field: InlineLabelField, hideError: true, serializer } },
-          };
-        },
-        { hidden: [], properties: {} } as any
-      ),
+      SCHEMA_URIS.map((key, index) => {
+        const [version] = VERSION_REGEX.exec(key) || [];
+        let selected = false;
+        if ($schema) {
+          selected = $schema && key === $schema;
+        } else {
+          selected = !index;
+        }
+        return {
+          text: formatMessage('Version {version}', { version }),
+          key,
+          selected,
+        };
+      }),
     []
   );
 
   useEffect(() => {
-    if (!value.$id) {
-      onChange({ $schema, $id: `${botName}-${uuid()}`, endpoints: [{}], name: botName, ...rest });
-    }
-  }, []);
+    const skillManifest = skillManifests.find(
+      (manifest) => manifest.content.$schema === ($schema || SCHEMA_URIS[0])
+    ) || {
+      content: {
+        $schema: $schema || SCHEMA_URIS[0],
+        $id: `${botName}-${uuid()}`,
```

Comment:
nit: There is a state called botNameIdentifier if you need something unique.

Reply:
use `botNameIdentifier `